### PR TITLE
Implement SMB Server Settings Resource

### PR DESCRIPTION
**Pull request by MangirishK**
# Description
SMB Server Settings Resource Implementation

# ISSUE TYPE
- Feature Pull Request

##### RESOURCE NAME
SMB Server Settings

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 80% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Unit tests
- [x] Acceptance tests

### CODE REVIEW

**Comment by MangirishK** [comment]
Please find the screenshots here for acceptance testing and coverage:

AT_Tests
AT_Coverage1
AT_Coverage2

**Comment by MangirishK** [comment]
run e2e test

**Review by P-Cao — APPROVED** [review]

File: powerscale/provider/smb_server_settings_resource.go [hunk 1]
```diff
@@ -0,0 +1,509 @@
...
+		Description: `This resource is used to manage the SMB Server Settings of PowerScale Array. We can Create, Update and Delete the SMB Server Settings using this resource.  
+		Note that, SMB Server Settings is the native functionality of PowerScale. When creating the resource, we actually load SMB Server Settings from PowerScale to the resource.`,
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed:            true,
+				Description:         "ID of SMB Server Settings.",
+				MarkdownDescription: "ID of SMB Server Settings.",
+			},
+			"scope": schema.StringAttribute{
+				Computed:            true,
```
Comment (RayLiu7) [inline powerscale/provider/smb_server_settings_resource.go:75]: Can we remove computed? Will scope be returned from API

Comment (MangirishK) [reply powerscale/provider/smb_server_settings_resource.go:75]: Scope is not returned from the API.
It is a query parameter for the API that can have 3 possible values "effective", "user" and "default".(If not specified, it is considered as "effective")

So, have kept 'scope' as both "Computed" and "Optional" as true in the schema.
If specified in the configuration file, it is picked up as it is.
Else if not specified, it is computed as "effective".

Comment (RayLiu7) [reply powerscale/provider/smb_server_settings_resource.go:75]: OK, got it. That means you set the scope in the state at last. Thanks for clarifying!

**Review by RayLiu7 — APPROVED** [review]

**Pull request merged**